go.mod

```diff
@@ -40,28 +39,28 @@ replace (
 	github.com/hashicorp/vault/api => github.com/hashicorp/vault/api v1.0.5-0.20200902155336-f9d5ce5a171a
 	github.com/hashicorp/vault/sdk => github.com/hashicorp/vault/sdk v0.1.14-0.20201116234512-b4d4137dfe8b
 	github.com/kubernetes-csi/external-snapshotter/v2 => github.com/kubernetes-csi/external-snapshotter/v2 v2.1.1-0.20200504125226-859696c419ff
-	k8s.io/api => k8s.io/api v0.20.6
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.20.6
-	k8s.io/apimachinery => k8s.io/apimachinery v0.20.6
-	k8s.io/apiserver => k8s.io/apiserver v0.20.6
-	k8s.io/cli-runtime => k8s.io/cli-runtime v0.20.6
-	k8s.io/client-go => k8s.io/client-go v0.20.6
-	k8s.io/cloud-provider => k8s.io/cloud-provider v0.20.6
-	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.20.6
-	k8s.io/code-generator => k8s.io/code-generator v0.20.6
-	k8s.io/component-base => k8s.io/component-base v0.20.6
-	k8s.io/component-helpers => k8s.io/component-helpers v0.20.6
-	k8s.io/controller-manager => k8s.io/controller-manager v0.20.6
-	k8s.io/cri-api => k8s.io/cri-api v0.20.6
-	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.20.6
-	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.20.6
-	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.20.6
-	k8s.io/kube-proxy => k8s.io/kube-proxy v0.20.6
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.20.6
-	k8s.io/kubectl => k8s.io/kubectl v0.20.6
-	k8s.io/kubelet => k8s.io/kubelet v0.20.6
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.20.6
-	k8s.io/metrics => k8s.io/metrics v0.20.6
-	k8s.io/mount-utils => k8s.io/mount-utils v0.20.6
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.20.6
+	k8s.io/api => k8s.io/api v0.21.2
```

Comment:
can these `replace` entries be removed now? The version in `require` is the same.

Reply:
@nixpanic 
I think we need to still `replace` entries, tried removing all the repeated entries but getting same error as below.
Am I missing/ doing something wrong ? :sweat: 
```
[rakshith@localhost ceph-csi]$ gd
diff --git a/go.mod b/go.mod
index 9c3ac4115..a5017850b 100644
--- a/go.mod
+++ b/go.mod
@@ -39,7 +39,6 @@ replace (
        github.com/hashicorp/vault/api => github.com/hashicorp/vault/api v1.0.5-0.20200902155336-f9d5ce5a171a
        github.com/hashicorp/vault/sdk => github.com/hashicorp/vault/sdk v0.1.14-0.20201116234512-b4d4137dfe8b
        github.com/kubernetes-csi/external-snapshotter/v2 => github.com/kubernetes-csi/external-snapshotter/v2 v2.1.1-0.20200504125226-859696c419ff
-       k8s.io/api => k8s.io/api v0.21.2
        k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.21.2
        k8s.io/apimachinery => k8s.io/apimachinery v0.21.2
        k8s.io/apiserver => k8s.io/apiserver v0.21.2
[rakshith@localhost ceph-csi]$ go mod download
go: k8s.io/kubernetes@v1.21.2 requires
        k8s.io/api@v0.0.0: reading k8s.io/api/go.mod at revision v0.0.0: unknown revision v0.0.0
```

Reply:
Replace is required to satisfy internal dependency what I have seen earlier was without this one, the latest dependency will get pulled, or the build will fail.

Reply:
yeah, replace is required,